pkggrp-ni-internal-deps: enable ni-grpc-device

rajendra-desai-ni opened the pull request:
ni-grpc-device has been updated to compile with grpc >= 1.60. This commit enables the recipe back into the package feeds.

### Summary of Changes

- enabled ni-grpc-device in packagegroup-ni-internal-deps.bb
- created patch files that include cmake build changes (from [this](https://github.com/ni/grpc-device/commit/0d1658ea666d25a8577f348d78645b591eb45cd8) commit) into the released version of ni-grpc-device (v2.6.0) for bitbake build.
- updated ni-grpc-device_git.bb to accept new supported arguments to build ni-grpc-device.


### Justification

The recipe was disabled due to compatibility issues as a part of [this](https://github.com/ni/meta-nilrt/pull/664) PR. Since we would now be shipping nilrt 11 to the customers, it becomes necessary to re-enable the recipe.

[AB#2640878](https://dev.azure.com/ni/DevCentral/_workitems/edit/2640878/)


### Testing

* [X] I have built the core package feed with this PR in place. (`bitbake packagefeed-ni-core`)

### Procedure

* [X] I certify that the contents of this pull request complies with the [Developer Certificate of Origin](https://github.com/ni/nilrt/blob/HEAD/docs/CONTRIBUTING.md#developer-certificate-of-origin-dco).

chaitu236 commented on recipes-ni/ni-grpc-device/ni-grpc-device_git.bb:
```diff
@@ -10,37 +10,37 @@ LIC_FILES_CHKSUM = "\
-	file://0002-CMakeLists-fixup-utf8cpp-library-link.patch \
-	file://0003-CMakeLists-use-native-python3-binaries.patch \
-	file://0001-CMakeLists-find-gpr-library.patch \
+	file://0001-CMakeLists_txt.patch \
```
Patch files are not usually created for individual file changes within same commit. A single patch file should ideally be used for all changes in a commit; and it's name should correspond to the title of the actual commit.

If you're using devtool to update this recipe, cherry-picking https://github.com/ni/grpc-device/commit/0d1658ea666d25a8577f348d78645b591eb45cd8 in the devtool workspace and `devtool update-recipe ..` should create the patch with correct name, etc.

If instead you want to manually create the patch, `git format-patch` can be used.

Please also set [Upstream-Status](https://github.com/ni/meta-nilrt/blob/nilrt/master/kirkstone/recipes-core/util-linux/util-linux/0001-skip-btrfs-tests-if-kernel-support-for-btrfs-is-miss.patch#L6).

If your intention in splitting the patch files was to not include changes form `.github/workflows/build_nilrt.yml`, you can instead just amend the commit after cherry-picking the patch in devtool workspace and revert changes in that file (You can make a note of this in the commit message just before sign-off tag like `[rajendra: Discarded upstream changes in .github/workflows/build_nilrt.yml]`).

rajendra-desai-ni replied:
Yeah, I had split up the patch files to not include .github changes and that's the reason the patch files were renamed based on file names instead of commit title.
I have now made all the suggested changes. Marking this comment as resolved.

rajendra-desai-ni committed:
pkggrp-ni-internal-deps: enable ni-grpc-device

ni-grpc-device has been updated to compile with grpc >= 1.60.
This commit enables the recipe back into the package feeds.

Signed-off-by: Rajendra Desai <rajendra.desai@ni.com>
